docs/spec/consensus/checkvalidators.md

```diff
@@ -0,0 +1,224 @@
+# Checking Validator Set (CheckVS)
+
+As part of the light client, the CheckVS procedure has to check, whether given
+two headers (whose heigts differ by more than 1), the LightClient can trust  the
+newer header under the assumption it trusted the old one.
+
+This document contains some math formulas. To ease reading, the file
+  /tendermint/docs/spec/pdfs/checkvalidators.pdf
+displays them correctly.
+
+## Definitions
+
+  * header fields
+    - $height$
+    - $bfttime$: the chain time when the header (block) was generated
+    - $V$: validator set containing validators.
+    - $nextV$: next validators
+  * $tp$: trusting period   
+  * for a time $t$, the predicate $correct(v,t)$ is true if the validator $v$
+    follows the protocol until time $t$ (we will see about recovery later).
+    Similarly we define $faulty(v,t)$.
+  * For each header $h$ it has locally stored, the LightClient stores whether
+    it trusts $h$. We write $trust(h) = true$, if this is the case.
+  * Validator fields. We will write a validator as a tuple $(v,p)$ such that
+    + $v$ is the identifier (we assume identifiers are unique in each validator set)
+    + $p$ is its voting power
+
+## LightClient Trusting Spec
+
+### LightClient Invariant
```

Comment:
I am not sure this is the best place to start the light client definition. Is this how we define full nodes? I think we just want the light client header acceptance algorithm to reflect what a full node would have done given the same set of messages (condensed into headers in the case of a light client).

In particular, this does not even hold for subsequent blocks (block 11 following block 10) if the validator set changes by more than 1/3, which is possible. It also doesn't capture fault-accountability (which we should be able to capture precisely, such that a light client could only diverge from a full node if a slashable signature event by at least a well-defined fraction of stake had occurred).